app/components/post_list/post/body/content/permalink_preview/permalink_preview.tsx

```diff
@@ -0,0 +1,238 @@
+            shadowRadius: 3,
+            elevation: 2,
+        },
+        header: {
+            flexDirection: 'row',
+            alignItems: 'center',
+            marginBottom: 8,
+        },
+        authorInfo: {
+            flex: 1,
+            marginLeft: 12,
+            flexDirection: 'row',
+            alignItems: 'center',
+        },
+        authorName: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100, 'SemiBold'),
+        },
+        timestamp: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75),
+            marginLeft: 8,
+        },
+        messageContainer: {
+            marginBottom: 8,
+            maxHeight: 20 * MAX_PERMALINK_PREVIEW_LINES,
+            overflow: 'hidden',
+        },
+        messageText: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100),
+            lineHeight: 20,
+        },
+
+        channelContext: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75),
+        },
+        channelName: {
+            color: theme.linkColor,
+            ...typography('Body', 75, 'SemiBold'),
+        },
+    };
+});
+
+const PermalinkPreview = ({
+    embedData,
+    author,
+    currentUser,
+    isMilitaryTime,
+    teammateNameDisplay,
+    isOriginPostDeleted,
+    location,
+}: PermalinkPreviewProps) => {
+    const theme = useTheme();
+    const serverUrl = useServerUrl();
+    const locale = useUserLocale();
+    const styles = getStyleSheet(theme);
+
+    const textStyles = getMarkdownTextStyles(theme);
+    const blockStyles = getMarkdownBlockStyles(theme);
+
+    const userId = embedData?.post?.user_id;
+
+    useEffect(() => {
+        if (userId && !author) {
+            fetchUsersByIds(serverUrl, [userId], false);
+        }
+    }, [userId, author, serverUrl]);
+
+    if (isOriginPostDeleted) {
+        return null;
+    }
+
+    const {
+        post,
+        channel_display_name,
+        channel_type,
+    } = embedData;
+
+    const truncatedMessage = useMemo(() => {
+        const message = post?.message;
+
+        if (!message || typeof message !== 'string') {
+            return '';
+        }
+
+        const cleanMessage = message.trim();
+        const lines = cleanMessage.split('\n');
+        if (lines.length > MAX_PERMALINK_PREVIEW_LINES) {
+            return lines.slice(0, MAX_PERMALINK_PREVIEW_LINES).join('\n...');
+        }
+
+        return cleanMessage;
+    }, [post?.message]);
+
+    const isEdited = useMemo(() => post && post.edit_at, [post]);
+
+    const authorDisplayName = useMemo(() => {
+        return displayUsername(author, locale, teammateNameDisplay);
+    }, [author, locale, teammateNameDisplay]);
+
+    const channelContextText = useMemo(() => {
+        const displayName = channel_type === 'D' ? authorDisplayName : channel_display_name;
+        return `~${displayName}`;
```

Comment:
if dm shouldn't it be `@` ?

Reply:
Webapp has ~ for DM. So keeping it the same in mobile as well.